Update README.md

K0ncsss opened the pull request:
ДЗ1

K0ncsss committed:
Update README.md

ДЗ1

VladimirBalun commented on README.md:
```diff
@@ -1 +1,155 @@
+# System Design социальной сети для путешественников для курса по System Design
+
+## Функциональные требования
+1.	Регистрация в соцети
```
Этого не было в ДЗ

VladimirBalun commented on README.md:
```diff
@@ -1 +1,155 @@
+3.	Публикация/удаление постов в личной ленте:
+    - С фото
+    - С текстом
+    - С указанием геометки места путешествия (**ВОПРОС**: как я это вижу, то будет интеграция со сторонним сервисом, в котором будет api по поиску геогр объектов (стран, городов, нас пунктов.) типо Яндекс карт. А наша система уже в БД будет хранить информацию полученную из Гео стороннего сервиса - наименование объекта (для вывода на UI), широту долготу возможно (ну это скорее всего больше на буду ...
```
Хорошо, но стоит это делать при записи, а не при чтении - так как система перекошена на чтение по нагрузку

K0ncsss replied:
Да, так и хотела, получаем при записи и сохраняем необходимые данные для выдачи поста на чтение после.
Что посоветуете почитать?

VladimirBalun replied:
Да, про почитать не скажу, где это описаны - мы в курсе будем еще это разбирать

VladimirBalun commented on README.md:
```diff
@@ -1 +1,155 @@
+10.	Поддержка поиска популярных мест для путешествий среди пользователей соц сети и просмотр постов с этих мест
+
+## Нефункциональные требования
+1.	Сезонность есть – в летние месяцы нагрузки больше (потребуется выделение ресурсов на этот период)
```
Хочется получить множитель сезона

K0ncsss replied:
Добавила, в 1,5 раза

VladimirBalun commented on README.md:
```diff
@@ -1 +1,155 @@
+4.	Кросс-девайсная синхронизация (мобильная и веб-версия системы)
+5.	Хранение данных пользователей и постов постоянное
+6.	Пользовательская активность:
+    - В среднем каждый пользователь публикует - 3 поста в сутки
```
Обычно намного меньше

K0ncsss replied:
Сделала 1 пост, поправила расчеты нагрузки по посту

VladimirBalun commented on README.md:
```diff
@@ -1 +1,155 @@
+### RPS
+**RPS-write**
+
+All_write_RPS:
```
Зачем общий считать?

K0ncsss replied:
Для понимания нагрузки на прокси-сервер, балансировщик? Может понадобится, чтобы понимать сколько ресурсов на них наложить для принятия запросов, а уже внутри системы они могут распределиться по подсистемам
Пока не убираю, жду ответ

VladimirBalun replied:
Чаще всего это не требуется, но если хочется - можно оставить

VladimirBalun commented on README.md:
```diff
@@ -1 +1,155 @@
+
+    10 000 000 * (3post + 3 react + 2 comment) /86 400= 80 000 000 / 86 400 ~ 930 RPS
+
+RPS_Post: 
```
Создание или что?

VladimirBalun commented on README.md:
```diff
@@ -1 +1,155 @@
+
+    10 000 000* 3 post /86400 ~ 350 RPS
+
+RPS_Reaction: 
```
Аналогично

VladimirBalun commented on README.md:
```diff
@@ -1 +1,155 @@
+
+    10 000 000* 3 react /86400 ~ 350 RPS
+
+RPS_Comment:
```
Аналогично

VladimirBalun commented on README.md:
```diff
@@ -1 +1,155 @@
+
+    10 000 000*2 comment /86400 ~ 230 RPS
+
+**RPS-read**
```
Увидел, но лучше явно с каждой операцией указывать, как по мне

VladimirBalun commented on README.md:
```diff
@@ -1 +1,155 @@
+
+**RPS-read**
+
+All_reed_RPS: 
```
Зачем общий?

VladimirBalun commented on README.md:
```diff
@@ -1 +1,155 @@
+
+    10 000 000 (1 pers_feed +5 follow_feed + 3 another_feed + 3 search_location + 3posts_location) / 86400 ~ 1740 RPS
+ 
+RPS_Posts_feed:
```
А выдача комментариев и лайков? Или они будут с постом отдаваться сразу? А если миллиона комментариев и лайков под постом? Сомнительно

K0ncsss replied:
C постом будет тянуться данные поста - текст, картинки, количество лайков. Просмотр комментариев под постом будет отдельным запросом, **добавила RPS + трафик на чтение комментариев под постом**

VladimirBalun commented on README.md:
```diff
@@ -1 +1,155 @@
+
+Traffic (Post):
+
+    	350RPS * 3 Mb = 1050 Mb/sec
```
Я бы считал отдельно трафик для медиа и метаинформации

K0ncsss replied:
Сделала отдельно траффик записи\чтения мета и медиа информации

VladimirBalun commented on README.md:
```diff
@@ -1 +1,155 @@
+
+    230 RPS * 25 Kb = 5750 Kb/sec ~ 6 Mb/sec
+  
+**All_traffic_write:**
```
Зачем общий?

K0ncsss replied:
Убрала

VladimirBalun commented on README.md:
```diff
@@ -1 +1,155 @@
+
+Traffic_Posts_feed: 
+
+    1050 RPS * (3Mb +5 Kb) * 20 (pagination) ~ 1050 RPS * 3Mb * 20 = 63000 Mb/sec
```
Аналогично про отдельный трафик

VladimirBalun commented on README.md:
```diff
@@ -1 +1,155 @@
+  - Traffic_search_location: 350 RPS * 1Kb (GEO_name) = 350 Kb/sec ~ 1 Mb/sec
+  - Traffic_posts_location: 350 RPS * (3Mb +5 Kb) * 20 (pagination) = 350 * 3 * 20 =21 000 Mb/sec
+
+**All_traffic_read:**
```
Зачем общий снова?

VladimirBalun commented on README.md:
````diff
@@ -1 +1,155 @@
+
+    1050 RPS * (3Mb +5 Kb) * 20 (pagination) ~ 1050 RPS * 3Mb * 20 = 63000 Mb/sec
+    
+Traffic_Search ```~ 21 000 Mb/sec```
````
Аналогично про отдельный трафик

K0ncsss replied:
Сделала отдельный

K0ncsss committed:
Update README.md

1 Добавила коэффициент сезонности
2. Изменила количества публикаций постов на 1 + скорректировала расчеты RPS и трафика
3. Добавила просмотр комментариев + RPS+трафик
4.  Удалила расчет общего трафика
5. Разделила трафик на мета и медиа

VladimirBalun commented on README.md:
```diff
@@ -1,35 +1,35 @@
 5.	Хранение данных пользователей и постов постоянное
 6.	Пользовательская активность:
-    - В среднем каждый пользователь публикует - 3 поста в сутки
+    - В среднем каждый пользователь публикует - 1 пост в сутки
```
Слишком часто все равно, как по мне

K0ncsss replied:
Сделаю 1 раз в неделю, замержу коммит

K0ncsss replied:
RPS и трафик тоже поправила всвязи со сменой частоты публикации

K0ncsss committed:
Update README.md

изменение частоты публикации поста - 1 раз в неделю
